src/avram/bulk_upsert.cr

```diff
@@ -14,10 +17,27 @@ class Avram::BulkUpsert(T)
 
   def args
     @records.map do |record|
-      record.changed_attributes.map(&.value)
+      changed_record_values(record, &.value)
     end.transpose
   end
 
+  private def build_permitted_fields(fields_arr)
+    record = @records.first
+    fields_arr.push(:created_at) if record.responds_to?(:created_at)
```

Comment:
What if a user includes this column in the changeset?

Reply:
👍 - will add a de-dup as well, though I think this approach is going to fall out of favor since I've learned there's no way to send mismatching tuples (compiler error)